control/config.go

```diff
@@ -84,6 +84,49 @@ func GetDefaultConfig() *Config {
 	}
 }
 
+// UnmarshalJSON unmarshals valid json into a Config.  An example Config can be found
+// at github.com/intelsdi-x/snap/blob/master/examples/configs/snap-config-sample.json
+func (c *Config) UnmarshalJSON(data []byte) error {
```

Comment:
We might want to add a comment before the definition of each of the Config structs that and new fields will need to also be added to the UnmarshalJSON methods.